usr.sbin/devinfo/devinfo.c

```diff
@@ -57,6 +57,19 @@ struct indent_arg
 	void	*arg;
 };
 
+
+static void
+print_indent(int n)
+{
+	char *s;
+
+	if (n < 1)
+		return;
+	asprintf(&s, "%*s", n, " ");
```

Comment:
This is going to be allocating and freeing this a lot (once per line).
And asprintf is an expensive way to do this.
And we don't need to be thread safe here.
It would be better to have a 1k static buffer. then this becomes:
n = min(n, sizeof(buffer) - 1); memset(buffer, ' ', n); buffer[n] = '\0';
Then there's no leaks, and 1k is so far beyond what you'll need it's not funny.

Reply:
I have to define min myself right? So like this then?
I'd rather not change the type of the argument to avoid the cast to size_t, because there is already some casting involved with the indentation that I don't properly understand.

```
#define	min(a, b)	((a < b) ? a : b)

static char buffer[1024];
...
static void
print_indent(int n)
{
	if (n < 1)
		return;

	n = min((size_t)n, sizeof(buffer) - 1);
	memset(buffer, ' ', n);
	buffer[n] = '\0';
	xo_emit("{Pa:%s}", buffer);
}
```

Reply:
You should include `sys/param.h` (there MIN() as a macro is already defined).
Also, you don't have to explicitly cast, as buffer size is smaller than INT_MAX anyway. With `-Wextra` you'll get a harmless compiler warning, though to silent it you have to cast (either with size_t) or with int e.g. `(int)sizeof(buffer) - 1`

Reply:
Ah, param.h it is then. It's compiled with -Werror., so the cast is needed.

Reply:
Fixed